motoko/minimal-counter-dapp/README.md

```diff
@@ -0,0 +1,212 @@
+
+```javascript
+public query func getCount() : async Nat {
+    return counter;
+};
+```
+
+#### reset()
+The `reset()` function resets the counter value to 0 and returns the value.
+
+```javascript
+public func reset() : async Nat {
+    counter := 0;
+    return counter;
+};
+```
+
+### Candid interface
+The Candid interface is automatically created, and it has a convenient UI, which provides an easy, user-friendly way to test the backend. The UI is also automatically generated, and the canister ID can be retrieved from the `dfx canister id <canister_name>` command.
+
+```bash
+$ dfx canister id __Candid_UI
+r7inp-6aaaa-aaaaa-aaabq-cai
+```
+
+**http://<candid_canister_id>.localhost:8000/?id=<backend_canister_id>**
+
+
+### Frontend
+The default project installed with `dfx new project_name` has an `index.html` file with page HTML and an `index.js` file with an implementation of the backend functions. These two files are modified in this example project to support the counter functionality, and the backend functions.
+
+#### HTML
+All HTML code is in the `src/minimal_dapp_assets/src/index.html` file, and most of the HTML is carried over from the default project. The button is kept and so is the section showing the result, just simplified.
+
+```html
+<!doctype html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <meta name="viewport" content="width=device-width">
+        <title>Minimal Dapp</title>
+        <base href="/">
+
+        <link type="text/css" rel="stylesheet" href="main.css" />
+    </head>
+    <body>
+        <img src="logo.png" alt="DFINITY logo" />
+        <section>
+            <button id="clickMeBtn">Click Me!</button>
+        </section>
+        <section id="counter"></section>
+    </body>
+</html>
+```
+
+#### Javascript
+Two eventlisteners are added to the JavaScript file, `src/minimal_dapp_assets/src/index.js`, the existing JavaScript file from the default project. One eventlistener is for detecting button clicks, and it's calling the `count()` function in the backend, and an eventlistener for page load is added to get the initial value of the counter with `getCount()`. The backend functions are imported through the Candid interface.
+
+```javascript
+import { minimaldapp } from "../../declarations/minimal_dapp";
+
+document.addEventListener('DOMContentLoaded', async function () {
+  const counter = await minimaldapp.getCount();
+  document.getElementById("counter").innerText = "Counter: " + counter;
+})
+
+document.getElementById("clickMeBtn").addEventListener("click", async () => {
+  const counter = await minimaldapp.count();
+  document.getElementById("counter").innerText = "Counter: " + counter;
+});
+```
+
+
+## Deployment
+The local network is started by running this command:
+
+```bash
+$ dfx start --background
+```
+
+When the local network is up and running, run this command to deploy the canisters:
+
+```bash
+$ dfx deploy
```

Comment:
This fails, in part because the project needs `webpack.config.js`.  I recommend getting up-to-date versions of these files from a project created with `dfx new`, and copying them into each of these examples:
- webpack.config.js
- package.json

Reply:
Done